From 2d5cda0eddc670949949ba1bc07f430715504a04 Mon Sep 17 00:00:00 2001
From: Jean Delvare <jdelvare@suse.de>
Date: Thu, 24 Apr 2025 11:41:57 +0200
Subject: [PATCH 24/45] Get rid of u64_range

Function u64_range is pretty useless now, better open-code the
operation.

Signed-off-by: Jean Delvare <jdelvare@suse.de>
---
 dmidecode.c | 2 +-
 util.c      | 6 ------
 util.h      | 1 -
 3 files changed, 1 insertion(+), 8 deletions(-)

diff --git a/dmidecode.c b/dmidecode.c
index 1acbefa..c52f975 100644
--- a/dmidecode.c
+++ b/dmidecode.c
@@ -3147,7 +3147,7 @@ static void dmi_mapped_address_extended_size(u64 start, u64 end)
 	if (start == end)
 		pr_attr("Range Size", "Invalid");
 	else
-		dmi_print_memory_size("Range Size", u64_range(start, end), 0);
+		dmi_print_memory_size("Range Size", end - start + 1, 0);
 }
 
 /*
diff --git a/util.c b/util.c
index 3f09ec5..7903c3e 100644
--- a/util.c
+++ b/util.c
@@ -261,9 +261,3 @@ void *mem_chunk(off_t base, size_t len, const char *devmem)
 
 	return p;
 }
-
-/* Returns end - start + 1, assuming start < end */
-u64 u64_range(u64 start, u64 end)
-{
-	return end - start + 1;
-}
diff --git a/util.h b/util.h
index 0d37f35..e6c5e49 100644
--- a/util.h
+++ b/util.h
@@ -27,4 +27,3 @@
 int checksum(const u8 *buf, size_t len);
 void *read_file(off_t base, size_t *len, const char *filename);
 void *mem_chunk(off_t base, size_t len, const char *devmem);
-u64 u64_range(u64 start, u64 end);
-- 
2.47.0

